### ISSUE TITLE
support assigning to globals under no-unused-vars and no-undef

### ISSUE THREAD

**Issue by tel** [eslint/eslint#2830]
I'm using eslint to check meteor files. It's often necessary (to comply with Meteor package conventions) to write something like the following

```javascript
React = require("react/addons");
```

If I try to write this with base `eslint` then it will complain, rightly, about having no undefined variables (`no-undef`). If I declare this variable as a (writeable) global

```javascript
/* global React:true */
React = require("react/addons");
```

then instead I get an error under `no-unused-vars`. This I believe is invalid as I *have* used this variable—by assigning to it.

The only way to eliminate this warning is to configure `no-unused-vars` to `"vars": "local"` but this means that if I declare a global and don't use it then that `/* global ... */` declaration lingers and defeats some of the documentation purpose of having them in the first place.

Is there a way to configure `no-unused-vars` to match this use case? Is there any other proper workaround?

**Comment by gyandeeps** [comment]
Have you tried looking into https://github.com/yannickcr/eslint-plugin-react

**Comment by tel** [comment]
I'm using it, but it is tangential to this. I just used `React` as an example because it was the one I was staring at at the time :)

**Comment by nzakas** [comment]
> This I believe is invalid as I have used this variable—by assigning to it.

The problem is that you aren't using React, you're defining it (using means you are reading its value). Assigning a value to a variable that is never read is exactly what this rule flags.

You can add globals in your configuration file, and those are not required to be used. http://eslint.org/docs/user-guide/configuring#specifying-globals

Another option is to just use a comment to disable the rule for that line. See http://eslint.org/docs/user-guide/configuring#configuring-rules for more on that.

**Comment by tel** [comment]
> The problem is that you aren't using React, you're defining it

While I agree with that rule in general, the fact that it's keyed with `global Foo:true` at the top of the file changes my feeling about that somewhat.

In any case, it's up to you guys. I'll try the conditional comments for the time being.
